exporter/sumologicexporter/filter.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumologicexporter
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	tracetranslator "go.opentelemetry.io/collector/translator/trace"
+)
+
+type filter struct {
+	regexes []*regexp.Regexp
+}
+
+// Fields represents concatenated metadata
+type Fields string
+
+func newFilter(fields []string) (filter, error) {
+	metadataRegexes := make([]*regexp.Regexp, len(fields))
+
+	for i, field := range fields {
+		regex, err := regexp.Compile(field)
+		if err != nil {
+			return filter{}, err
+		}
+
+		metadataRegexes[i] = regex
+	}
+
+	return filter{
+		regexes: metadataRegexes,
+	}, nil
+}
+
+// filter returns map of strings which matches at least one of the filter regexes
+func (f *filter) filter(attributes pdata.AttributeMap) map[string]string {
```

Comment:
Consider `filterIn` to be symmetric with `filterOut`?

Reply:
That makes sense. Updated